openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py

```diff
@@ -0,0 +1,236 @@
+
+        # Initialize the DelayedRequester and all necessary Media Stores.
+        self.delayed_requester = DelayedRequester(self.delay)
+        self.media_stores = self.init_media_stores()
+        self.date = date
+
+    def init_media_stores(self) -> dict[str, MediaStore]:
+        """
+        Initialize a media store for each media type supported by this
+        provider.
+        """
+        media_stores = {}
+
+        for media_type, provider in self.providers.items():
+            StoreClass = get_media_store_class(media_type)
+            media_stores[media_type] = StoreClass(provider)
+
+        return media_stores
+
+    def ingest_records(self, **kwargs) -> None:
+        """
+        The main ingestion function that is called during the `pull_data` task.
+
+        Optional Arguments:
+        **kwargs: Optional arguments to be passed to `get_next_query_params`.
+        """
+        should_continue = True
+        record_count = 0
+        query_params = None
+
+        logger.info(f"Begin ingestion for {self.__class__.__name__}")
+
+        while should_continue:
+            query_params = self.get_next_query_params(query_params, **kwargs)
+
+            batch, should_continue = self.get_batch(query_params)
+
+            if batch and len(batch) > 0:
+                record_count = self.process_batch(batch)
+                logger.info(f"{record_count} records ingested so far.")
+            else:
+                logger.info("Batch complete.")
+                should_continue = False
+
+            if self.limit and record_count >= self.limit:
+                logger.info(f"Ingestion limit of {self.limit} has been reached.")
+                should_continue = False
+
+        self.commit_records()
+
+    @abstractmethod
+    def get_next_query_params(self, old_query_params: Optional[Dict], **kwargs) -> Dict:
+        """
+        Given the last set of query params, return the query params
+        for the next request. Depending on the API, this may involve incrementing
+        an `offset` or `page` param, for example.
+
+        Required arguments:
+        old_query_params: Dictionary of query string params used in the previous
+                          request. If None, this is the first request.
+        **kwargs:         Optional kwargs passed through from `ingest_records`.
+
+        """
+        pass
+
+    def get_batch(self, query_params: Dict) -> Tuple[Optional[List], bool]:
+        """
+        Given query params, request the next batch of records from the API and
+        return them in a list.
+
+        Required Arguments:
+        query_params: query string parameters to be used in the request
+
+        Return:
+        batch:           list of records in the batch
+        should_continue: boolean indicating whether ingestion should continue after
+                         this batch has been processed
+        """
+        batch = None
+        should_continue = True
+
+        try:
```

Comment:
Some provider scripts also set `headers`:
https://github.com/WordPress/openverse-catalog/blob/dc8a68ba3b7784bc460b5e4758acfd14c42ddc15/openverse_catalog/dags/providers/provider_api_scripts/science_museum.py#L98-L99

Reply:
Sorry, I found the answer: You can use the class attribute on a child DataIngester class :)